Demo: Different behavior in `Package.resolved` generation between Xcode 15.4 and `xcodebuild`

mokagio opened the pull request:
In short:

- Xcode 15.4 "Resolve Package Version" will generate `podcasts.xcworkspace/xcshareddata/swiftpm/Package.resolved`
- `xcodebuild -resolvePackageDependencies` will generated a `Package.resolved` in both workspace and project `xcshareddata` folders

mokagio committed:
Track result of `xcodebuild -resolvePackageDependencies`

mokagio committed:
Test – Delete both `Package.resolved`

mokagio committed:
Regenerate `Package.resolved` via Xcode 15.4

Xcode > File > Packages > Resolve Package Versions

Notice `Package.resolved` that was re-generated is the one in
the `xcworkspace`

mokagio committed:
Revert "Regenerate `Package.resolved` via Xcode 15.4"

This reverts commit ea07eda12dbd717dd436f05d5d7aa70de745404f.

mokagio committed:
Regenrate `Package.resolved` via `xcodebuild`

Notice that `xcodebuild` generates `Package.resolved` in both:

- `podcasts.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved`
- `podcasts.xcworkspace/xcshareddata/swiftpm/Package.resolved`

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Revert "Regenrate `Package.resolved` via `xcodebuild`"

This reverts commit e05256a1ede06f1790abc3d643022ca8f793512e.

mokagio committed:
Track result of `xcodebuild -resolvePackageDependencies` with workspace

Full call

```
xcodebuild -resolvePackageDependencies \
        -workspace podcasts.xcworkspace \
        -scheme pocketcasts
```

Passing a scheme is required, even though the option is not documented
in `xcodebuild --help`.

The result is a `Package.resolved` generated only in the worspace
`xcshareddata` folder

mokagio committed:
Revert "Track result of `xcodebuild -resolvePackageDependencies` with workspace"

This reverts commit f0d5d31d5ed4522e21a892b6fc9f96d1bf0e51a4.

mokagio committed:
Track `xcodebuild -resolvePackageDependencies` with `-project` result

The result is two `Package.resolved`, on for the project and the other
for the workspace.

Full command:

```
xcodebuild -resolvePackageDependencies -workspace <workspacename>
```
